packages/jsActions/mobile-resources-native/src/file-download/DownloadFile.ts

```diff
@@ -0,0 +1,70 @@
+// This file was generated by Mendix Studio Pro.
+//
+// WARNING: Only the following code will be retained when actions are regenerated:
+// - the import list
+// - the code between BEGIN USER CODE and END USER CODE
+// - the code between BEGIN EXTRA CODE and END EXTRA CODE
+// Other code you write will be lost the next time you deploy the project.
+import { Platform } from "react-native";
+import RNBlobUtil from "react-native-blob-util";
+import FileViewer from "react-native-file-viewer";
+
+// BEGIN EXTRA CODE
+function formatMendixFileUrl(file: mendix.lib.MxObject): string {
+    return `${mx.remoteUrl}file?guid=${file.getGuid()}&changedDate=${
+        file.get("changedDate") ?? ""
+    }&name=${encodeURIComponent(file.get("Name"))}`;
+}
```

Comment:
Can we use `mx.data.getDocumentUrl` method instead?

Reply:
This made me realize something fundamental (thanks @r0b1n 😄): if I'm not mistaken, the offline data backend will produce a URL which points to the local file system. Which begs the question: if the `MxObject` has already been synced to the offline device (because how else would you have the `MxObject` in your hands here), why would you still need to download the file from the remote server?

Reply:
Done. Can you check again? @r0b1n @tamelbrun

Reply:
✨